src/components/SafeListSidebar/SafeList/index.tsx

```diff
@@ -45,70 +42,96 @@ const useStyles = makeStyles({
       paddingLeft: '44px',
     },
   },
-  noIcon: {
-    visibility: 'hidden',
-    width: '28px',
-  },
 })
 
 type Props = {
-  currentSafeAddress: string | undefined
-  safes: SafeRecordWithNames[]
-  ownedSafes: string[]
   onSafeClick: () => void
 }
 
-const isAddressAdded = (addedSafes: SafeRecordWithNames[], address: string): boolean => {
-  return addedSafes.some((safe) => sameAddress(safe.address, address))
+const isAddressAdded = (addedSafes: SafeRecordWithNames[], address: string): boolean =>
+  addedSafes.some((safe) => sameAddress(safe.address, address))
+
+const getSafesStoredLocally = (
+  networkId: ETHEREUM_NETWORK,
+  safes: SafeRecordWithNames[],
+  ownedSafes: string[],
+): SafeRecordWithNames[] => {
+  let otherSafes: SafeRecordWithNames[] = []
+
+  const name = getNetworkName(networkId)
+  const safeString = localStorage.getItem(getSafesKey(name))
+
+  if (safeString) {
+    try {
+      Object.values(JSON.parse(safeString)).forEach((parsedSafe: any) => {
+        const isAddedSafe = isAddressAdded(safes, parsedSafe.address)
+        const isOwnedSafe = ownedSafes.includes(parsedSafe.address)
+        if (isAddedSafe || isOwnedSafe) return
+
+        otherSafes = [...otherSafes, parsedSafe]
+      })
+    } catch {}
+  }
+
+  return otherSafes
 }
 
-export const SafeList = ({ currentSafeAddress, onSafeClick, safes, ownedSafes }: Props): React.ReactElement => {
+export const SafeList = ({ onSafeClick }: Props): React.ReactElement => {
   const classes = useStyles()
-  const ownedSafesExpanded = ownedSafes.some((address) => {
-    return address === currentSafeAddress && !isAddressAdded(safes, address)
-  })
-
-  const getLink = (address: string): React.ReactElement =>
-    sameAddress(currentSafeAddress, address) ? (
-      <StyledIcon type="check" size="md" color="primary" />
-    ) : (
-      <div className={classes.noIcon}>placeholder</div>
-    )
+  const networks = getNetworks()
+  const currentSafeAddress = useSelector(safeAddressFromUrl)
+  const safes = useSelector(sortedSafeListSelector).filter((safe) => !safe.loadedViaUrl)
+  const ownedSafes = useOwnerSafes()
+  const safesNotAdded = ownedSafes.filter((address) => !isAddressAdded(safes, address))
```

Comment:
It seems logical to only display the not added safes but we specifically made it display _all_ the owned safes as per CEO's request.

Reply:
Do we want to show anything underneath the network heading, or just leave it blank? Maybe add Safe perhaps?

Reply:
The owned safes menu should just remain as it is.

Reply:
![image](https://user-images.githubusercontent.com/20442784/133757581-e6e65645-0f11-4a00-9cd1-d53c14a458a2.png)

Like this?

Reply:
Or perhaps with a message along the lines of: `No added Safes on ${network}`

Reply:
My point was that it should display _all_ owned safes, not just not added. See #2657 for context.

Reply:
All safes are shown now.